pkg/scheduler/helper.go

```diff
@@ -84,3 +90,20 @@ func getAffinityIndex(affinities []policyv1alpha1.ClusterAffinityTerm, observedN
 	}
 	return 0
 }
+
+// getConditionByError returns condition by error type, bool to indicate if ignore this error.
+func getConditionByError(err error) (metav1.Condition, bool) {
+	if err == nil {
+		return util.NewCondition(workv1alpha2.Scheduled, workv1alpha2.BindingReasonSuccess, successfulSchedulingMessage, metav1.ConditionTrue), true
+	}
+
+	var fitErr *framework.FitError
+	if errors.As(err, &fitErr) {
+		return util.NewCondition(workv1alpha2.Scheduled, workv1alpha2.BindingReasonNoClusterFit, err.Error(), metav1.ConditionFalse), true
```

Comment:
nit: if return true, scheduler will not retry by exponential backoff algorithm which could be unexpected from before.

Reply:
no, I just keep it same as before.
https://github.com/karmada-io/karmada/blob/8388314fbe80ff3bcbf72c2223dd3ba8f167cf50/pkg/scheduler/scheduler.go#L452-L464